gordo_components/client/client.py

```diff
@@ -126,6 +127,33 @@ def __init__(
         self.target = target
         self.machines = self.get_machines()
 
+    @property
+    def session(self):
+        return self._session
+
+    @session.setter
+    def session(self, session: requests.Session):
+        session.get = self._expired_revision_check(session.get)  # type: ignore
+        session.post = self._expired_revision_check(session.post)  # type: ignore
+        self._session = session
+
+    def _expired_revision_check(self, f):
+        """
+        Wrap a request function to check for 410 status codes, and update
+        the current revision if so.
+        """
+
+        @wraps(f)
+        def _wrapper(*args, **kwargs):
+            resp = f(*args, **kwargs)
+            if resp.status_code == 410:
+                self.revision = None  # Trigger updating revision
```

Comment:
This seems like it may have to do with the injecting of the revision in the headers of success responses. Maybe we could have some detailed docs on what revision business we expect in the various response headers.